### ISSUE TITLE
Add the masked diffusion model for pretrained baler autoencoder

### ISSUE THREAD

**Issue by neogyk** [baler-collaboration/baler#389]
The baler autoencoder is trained in the traditional end-end fashion, but it's possible to improve the current perfromance score with more advanced models and training techniques.
The masked diffusion model shows fast and accurate results by generating data of different modalities, so it's a nice candidate for research.
Things that necessary to add:

- the noise scheduler
- diffusion forward and reverse processes
- loss function for the diffusion model
- time embedding
- Test it using the image and tabular data

[Example of the MDM implementation](https://github.com/ML-GSAI/eMIGM).
